apps/labs/posts/blas-lapack-packaging.md

```diff
@@ -0,0 +1,372 @@
+  <img src="/posts/blas-lapack-packaging/lapack-libs.png"
+    width="893" height="198" alt="A diagram showing correspondence
+    between Netlib LAPACK, BLIS and OpenBLAS libraries. Netlib BLAS
+    consists of four libraries: blas, cblas, lapack and lapacke. cblas
+    is the C API to blas, whereas lapacke is the C API to lapack.
+    Additionally, lapack uses blas. BLIS is a single library that
+    corresponds both to blas and cblas. lapack can optionally use it
+    instead of blas. OpenBLAS is a single library that corresponds
+    to all four Netlib libraries." />
+  <figcaption>Fig. 1. Library correspondence between Netlib LAPACK, BLIS and OpenBLAS</figcaption>
+</figure>
+</div>
+
+The modern version of the Netlib LAPACK
+implementation features five libraries: `blas`, `cblas`, `lapack`,
+`lapacke` and `tmglib`. The `blas` library provides low-level routines
+for linear algebra, such as vector and matrix arithmetic. The `lapack`
+library builds on these routines to provide higher-level functionality,
+for example solving systems of linear equations. Both of these
+libraries are written in the Fortran programming language, and therefore
+provide a programming interface specific to Fortran. The `cblas`
+and `lapacke` libraries provide C-style interfaces to these routines.
+Finally, `tmglib` is a library of routines used for testing, far less
+known and rarely used outside the project.
+
+Netlib LAPACK splits not only the actual libraries, but also the related
+development files (headers, `pkg-config` files), therefore encouraging
+distributions to split the relevant packages as well. Other
+implementations often combine all the provided interfaces into a single
+library, such as `openblas`.
+
+Sometimes invidual interfaces are optional or not implemented at all.
+In OpenBLAS builds, CBLAS, LAPACK and LAPACKE interfaces can
+be disabled, though doing that is discouraged to avoid introducing
+compatibility issues. BLIS
+does not implement LAPACK at all, while providing BLAS and CBLAS
+interfaces (the latter being optional). It is usually combined
+with Netlib LAPACK; [libflame](https://github.com/flame/libflame)
+is being developed as a LAPACK counterpart to BLIS, but it is
+not commonly used at the time of writing.
+
+## LP64 and ILP64 interfaces
+
+<div style={{ textAlign: "center" }}>
+<figure style={{width: 'auto', margin: '0 2em', display: 'inline-block', verticalAlign: 'top'}}>
+  <img src="/posts/blas-lapack-packaging/lapack-lp64-ilp64.png"
+    width="746" height="173" alt="A diagram illustrating different
+    approaches to LP64 and ILP64 libraries. It is split into LP64,
+    ILP64 and 'both' parts. The Netlib lapack and openblas libraries
+    represent the pure LP64 libraries. The Netlib lapack (w/ ILP64)
+    and mkl_intel_lp64 libraries provide both LP64 and ILP64 interfaces.
+    The Netlib lapack64, openblas64, openblas64_ (with underscore)
+    and mkl_intel_ilp64 libraries provide only ILP64 interface." />
```

Comment:
This looks good now. I thought the `mkl_intel_lp64` was wrong, but it's indeed "both" (a bit counter-intuitively). So 👍🏼